llama_index/indices/vector_store/base.py

```diff
@@ -259,6 +259,9 @@ def build_index_from_nodes(
             VectorStoreIndex only stores nodes in document store
             if vector store does not store text
         """
+        nodes = [
```

Comment:
honest question should we just explicitly throw an error? wonder if this would make it harder for user to realize they have empty nodes

Reply:
Yea right now it throws a very non-obvious error.

Maybe throwing a more informative error is a better idea